pkg/kapp/diff/explicit_versioned_ref.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"fmt"
+	"reflect"
+
+	"gopkg.in/yaml.v2"
+)
+
+type VersionedRefDesc struct {
+	Namespace  string `yaml:"namespace"`
+	APIVersion string `yaml:"apiVersion"`
+	Kind       string `yaml:"kind"`
+	Name       string `yaml:"name"`
+}
+
+type ExplicitVersionedRef struct {
+	Resource   VersionedResource
+	Annotation string
+}
+
+const (
+	explicitReferenceKey       = "kapp.k14s.io/versioned-explicit-ref"
+	explicitReferenceKeyPrefix = "kapp.k14s.io/versioned-explicit-ref."
+)
+
+func NewExplicitVersionedRef(res VersionedResource, annotation string) *ExplicitVersionedRef {
+	return &ExplicitVersionedRef{res, annotation}
+}
+
+// Returns true if the resource is referenced by the annotation
+func (e *ExplicitVersionedRef) IsReferenced() (bool, error) {
+	reference := VersionedRefDesc{}
+	err := yaml.Unmarshal([]byte(e.Annotation), &reference)
+	if err != nil {
+		return false, fmt.Errorf("Error unmarshalling versioned reference: %s", err)
+	}
+
+	if reference.APIVersion == "" || reference.Kind == "" || reference.Name == "" {
+		return false, fmt.Errorf("Explicit reference error: apiVersion, kind and name are required values in an explicit versioned reference")
```

Comment:
Nit: Maybe we can have something like this instead of mentioning `Explicit reference` two times - 
```suggestion
		return false, fmt.Errorf("Explicit versioned reference error: apiVersion, kind and name are required values")
```

Reply:
Doesn't hurt to be more specific. Could certainly make this change!